filters/openpolicyagent/openpolicyagent.go

```diff
@@ -764,6 +757,27 @@ func (opa *OpenPolicyAgentInstance) DistributedTracing() opatracing.Options {
 	return buildTracingOptions(opa.registry.tracer, opa.bundleName, opa.manager)
 }
 
+// CreatePreparedQueryOnce is an implementation of the envoyauth.EvalContext interface
+func (opa *OpenPolicyAgentInstance) CreatePreparedQueryOnce(opts envoyauth.PrepareQueryOpts) (*rego.PreparedEvalQuery, error) {
+	opa.preparedQueryDoOnce.Do(func() {
+		regoOpts := append(opts.Opts, rego.DistributedTracingOpts(opa.DistributedTracing()))
+
+		prepareOpts := opts.PrepareOpts
+
+		if opa.registry.preevaluationOptimization {
```

Comment:
I think this is accidental and should not be part of this PR

Reply:
Thanks. Addressed.